tests/gui/models/selection-state.spec.js

```diff
@@ -425,10 +448,75 @@ describe('Browser: SelectionState', function() {
           }).to.throw('Invalid image path: 123');
         });
 
+        it('should throw if no extension', function() {
+          m.chai.expect(function() {
+            SelectionStateModel.setImage({
+              path: 'foo.img',
+              size: {
+                original: 999999999,
+                final: {
+                  estimation: false,
+                  value: 999999999
+                }
+              }
+            });
+          }).to.throw('Missing image extension');
+        });
+
+        it('should throw if extension is not a string', function() {
+          m.chai.expect(function() {
+            SelectionStateModel.setImage({
+              path: 'foo.img',
+              extension: 1,
+              size: {
+                original: 999999999,
+                final: {
+                  estimation: false,
+                  value: 999999999
+                }
+              }
+            });
+          }).to.throw('Invalid image extension: 1');
+        });
+
+        it('should throw if the extension doesn\'t match the path and there is no archive extension', function() {
+          m.chai.expect(function() {
+            SelectionStateModel.setImage({
+              path: 'foo.img',
+              extension: 'iso',
+              size: {
+                original: 999999999,
+                final: {
+                  estimation: false,
+                  value: 999999999
+                }
+              }
+            });
+          }).to.throw('Missing image archive extension');
+        });
+
+        it('should throw if the extension doesn\'t match the path and the archive extension is not a string', function() {
+          m.chai.expect(function() {
+            SelectionStateModel.setImage({
+              path: 'foo.img',
+              extension: 'iso',
+              archiveExtension: 1,
+              size: {
+                original: 999999999,
+                final: {
+                  estimation: false,
+                  value: 999999999
+                }
+              }
+            });
+          }).to.throw('Invalid image archive extension: 1');
+        });
```

Comment:
I added a test case for the first one. The other case is a bit tricky to test. If a file is `foo.zip` and contains a file called `bar.img` inside, then the file path doesn't reflect the internal file's extension.